tests/build/build.go

```diff
@@ -123,6 +128,241 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
+
+					err := json.Unmarshal(statusBytes, &buildStatus)
+					if err != nil {
+						GinkgoWriter.Printf("cannot unmarshal build status: %v\n", err)
+						return false, err
+					}
+
+					if buildStatus.PaC != nil {
+						GinkgoWriter.Printf("state: %s\n", buildStatus.PaC.State)
+						GinkgoWriter.Printf("mergeUrl: %s\n", buildStatus.PaC.MergeUrl)
+						GinkgoWriter.Printf("errId: %d\n", buildStatus.PaC.ErrId)
+					} else {
+						GinkgoWriter.Println("build status does not have PaC field")
+					}
+
+					return buildStatus.PaC != nil && buildStatus.PaC.State == "enabled" && buildStatus.PaC.MergeUrl != "" && buildStatus.PaC.ErrId == 0, nil
+				}, timeout, interval).Should(BeTrue(), fmt.Errorf("build status has unexpected content"))
+
+				Expect(component.Annotations[controllers.BuildRequestAnnotationName]).To(BeEmpty(), "request annotation was not deleted")
+			})
+
+			It("build pipeline is triggered after merge", func() {
+				var pipelinerunPullSha string
+				var mergeResult *github.PullRequestMergeResult
+
+				Eventually(func() (bool, error) {
+					prs, err := f.AsKubeAdmin.CommonController.Github.ListPullRequests(helloWorldComponentGitSourceRepoName)
+					if err != nil {
+						GinkgoWriter.Printf("cannot list pull requests: %v\n", err)
+						return false, err
+					}
+
+					for _, pr := range prs {
+						if pr.Head.GetRef() == annotationsBranchName {
+							prNumber = pr.GetNumber()
+							pipelinerunPullSha = pr.Head.GetSHA()
+							return true, nil
+						}
+					}
+					return false, nil
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for init PaC PR to be created")
+
+				mergeResult, err := f.AsKubeAdmin.CommonController.Github.MergePullRequest(helloWorldComponentGitSourceRepoName, prNumber)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Eventually(func() bool {
+					pipelineRunPull, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(annotationsComponentName, applicationName, testNamespace, pipelinerunPullSha)
+					if err != nil {
+						GinkgoWriter.Printf("PipelineRun on pull request has not been created yet, expecting sha `%s`", pipelinerunPullSha)
+						return false
+					}
+					pipelineRunPush, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(annotationsComponentName, applicationName, testNamespace, mergeResult.GetSHA())
+					if err != nil {
+						GinkgoWriter.Printf("PipelineRun on push has not been created yet, expecting sha `%s`", mergeResult.GetSHA())
+						return false
+					}
+					return pipelineRunPull.HasStarted() && pipelineRunPush.HasStarted()
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for the PipelineRuns to start")
+
+				component, err = f.AsKubeAdmin.HasController.GetComponent(annotationsComponentName, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(component.Annotations[controllers.BuildRequestAnnotationName]).To(BeEmpty(), "request annotation was not deleted")
```

Comment:
similarly observing this checking component annotation code pattern multiple times with this change.. will it be worth adding a new function named CheckBuildRequestAnnotationIsEmpty and move the functionality there